src/codegen/generators/expression_generator.rs

```diff
@@ -68,13 +68,25 @@ pub struct ExpressionCodeGenerator<'a, 'b> {
 /// context information to generate a parameter
 #[derive(Debug)]
 struct CallParameterAssignment<'a, 'b> {
-    /// the assignmentstatement in the call-argument list (a:=3)
+    /// The named argument node of the function call (e.g. `foo(in := 3)`)
     assignment: &'b AstNode,
-    /// the name of the function we're calling
+
+    /// The name of the POU being called
     function_name: &'b str,
-    /// the position of the argument in the POU's argument's list
-    index: u32,
-    /// a pointer to the struct instance that carries the call's arguments
+
+    /// The position of the parameter in the POUs variable declaration list.
+    /// See also [`StatementAnnotation::Argument::position`].
+    position: u32,
+
+    /// The depth between where the parameter is declared versus where it is being called from.
+    /// See also [`StatementAnnotation::Argument::depth`].
+    depth: u32,
+
+    /// The name of the POU where the parameter is declared
+    /// See also [`StatementAnnotation::Argument::pou`].
+    pou_parameter: &'b str,
```

Comment:
If it weren't for the doc-comment, I'd have no idea that `pou_parameter` refers to the name of the POU. How do you feel about something like `declaring_pou` or `source_pou`?